### PR TITLE
	[FLINK-5084] Replace Java Table API integration tests by unit tests	

### PR DESCRIPTION
**Pull request by mtunique**
I replace the Java Table API ITCases by tests that check that the parsing step produces a valid logical plan.
Before I delete the ITCases, I have check there are UTCases in scala.

I left the Java special ITCases.

### CODE REVIEW

**Comment by fhueske** [comment]
Thanks for working on this @mtunique!

I would like to suggest the following. All test classes in `./test/java/org/apache/flink/api/java/batch/table` are split into a validation and a plan test class. All tests are implemented in Scala and moved to the `./test/scala/` directory.

- The validation tests contain the test methods which check for failing validation. These tests are all unit tests and are named like `CalcValidationTest`.
- The plan test compare the logical plans of the string Table API and the expression Table API. These are also unit tests and named like `CalcPlanTest`. I would not merge these tests with the execution tests of the expression Table API but keep them separate.

The file `./test/java/org/apache/flink/api/java/batch/ExplainTest` can be removed. It checks exactly the same as the Scala version of this test.

What do you think?

Best, Fabian

**Comment by mtunique** [comment]
Thanks @fhueske .
I agree very much.
It is necessary to split the tests into a validation and a plan test class.
I will follow the suggestions.

**Comment by mtunique** [comment]
@fhueske 
I have finish there tasks:
- remove `./test/java/org/apache/flink/api/java/batch/ExplainTest`
- All tests in `test/java/org/apache/flink/api/java/batch/table` are implemented in Scala and moved to the ./test/scala/ directory.
- Move validation tests into new files.
- Add plan tests (some have the java version and some are new).

It is time to review these codes.

**Comment by mtunique** [comment]
Hi @fhueske ,

I have fixed the comments.
Maybe you can review the changes, then I rebase the master branch and refactor the directory structure.Maybe this can be easy to review by two steps?

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/SortValidationTest.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.apache.flink.api.scala.util.CollectionDataSets
+import org.apache.flink.api.scala.{ExecutionEnvironment, _}
+import org.apache.flink.api.table.{Row, TableEnvironment, ValidationException}
+import org.apache.flink.test.util.MultipleProgramsTestBase.TestExecutionMode
+import org.junit._
+
+class SortValidationTest(
+  mode: TestExecutionMode,
+  configMode: TableConfigMode)
+  extends TableProgramsTestBase(mode, configMode) {
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/SortValidationTest.scala]: `TableProgramsTestBase` should be removed.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/CalcITCase.scala [hunk 2]
```diff
@@ -109,36 +110,6 @@ class CalcITCase(
     TestBaseUtils.compareResultAsText(results.asJava, expected)
   }
 
-  @Test(expected = classOf[ValidationException])
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/CalcITCase.scala]: The test method `testAliasStarException()` should be moved to `CalcValidationTest` and be split up into individual methods.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/SortValidationTest.scala (hunk 1 above)
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/table/SortValidationTest.scala]: `TablePRogramsTestBase` should be removed.

**Comment by mtunique** [comment]
@fhueske I have finished all about comments and rebase.

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/table/SetOperatorsValidationTest.scala [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+package org.apache.flink.table.api.scala.batch.table
+
+import org.apache.flink.api.scala._
+import org.apache.flink.api.scala.util.CollectionDataSets
+import org.apache.flink.table.api.scala._
+import org.apache.flink.table.api.{TableEnvironment, ValidationException}
+import org.junit._
+
+class SetOperatorsValidationTest {
```
Comment (fhueske) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/batch/table/SetOperatorsValidationTest.scala]: This class should be moved into the `validation` package.
Can be done before merging

**Comment by fhueske** [comment]
Thanks for the update and rebasing @mtunique!

PR is good to merge (one file should be moved, see comment above).

**Comment by mtunique** [comment]
@fhueske Done. Thanks.

**Comment by twalthr** [comment]
I will have a final look over the code and merge this.

**Pull request closed without merging**